config/configuration.go

```diff
@@ -30,9 +31,12 @@ var (
 	// DefaultUnixLogLocation is the primary location to find log files
 	DefaultUnixLogLocation = "/var/log/cloudflared"
 
+	// DefaultXDGConfigLocation is the primary user location to find a config file
+	DefaultXDGConfigLocation = xdg.ConfigHome + "/cloudflared"
```

Comment:
Can we do this without adding a vendor to just bring in this variable?

Reply:
what do you men by that?

We need to specify a folder name for within `${XDG_CONFIG_HOME:-$HOME/.config}`
It will result in something similar to:
```
~/.config/cloudflared
├── XXX-XXX-XXX-XXX-XXX.json
└── cert.pem
```

Reply:
The two comments are connected. You are relying on this library to populate the xdgconfig location prefix and my suggestion is to do that in the PR here and not bloat the the package with a library for just this logic.

Reply:
Just asking should the behavior also be implemented on OSX and Windows?
The appropriate dirs are `~/Library/Application Support` and `LocalAppData` but I don't know how popular they are?